File: src/main/java/com/viartemev/requestmapper/annotations/spring/extraction/PsiAnnotationMemberValueExtractor.java
package com.viartemev.requestmapper.annotations.spring.extraction;

import com.intellij.psi.PsiAnnotationMemberValue;
import org.apache.commons.lang.StringUtils;
import org.jetbrains.annotations.NotNull;

import java.util.Collections;
import java.util.List;

import static com.viartemev.requestmapper.utils.CommonUtils.unquote;

public class PsiAnnotationMemberValueExtractor implements PsiAnnotationValueExtractor<PsiAnnotationMemberValue> {
    @Override
    public List<String> extract(@NotNull PsiAnnotationMemberValue value) {
        return StringUtils.isNotEmpty(value.getText()) ?
               Collections.singletonList(unquote(value.getText())) :
               Collections.emptyList();
    }
}

File: src/main/kotlin/com/viartemev/requestmapper/annotations/spring/extraction/PsiAnnotationMemberValueExtractor.kt
package com.viartemev.requestmapper.annotations.spring.extraction

import com.intellij.psi.PsiAnnotationMemberValue
import com.viartemev.requestmapper.utils.unquote

class PsiAnnotationMemberValueExtractor : PsiAnnotationValueExtractor<PsiAnnotationMemberValue> {
    override fun extract(value: PsiAnnotationMemberValue): List<String> {
        return if (value.text.isNotBlank()) listOf(value.text.unquote()) else emptyList<String>()
    }
}
